app/assets/javascripts/compute_resource.js

```diff
@@ -78,7 +78,7 @@ function ovirt_templateSelected(item){
         $('[id$=_cores]').val(result.cores);
         $('#network_interfaces').children('.fields').remove();
         $.each(result.interfaces, function() {add_network_interface(this);});
-        $('#storage_volumes').children('.fields').remove();
+        $('#storage_volumes .children_fields >.fields').remove();
```

Comment:
This will also remove volumes created manually but otherwise it's a lot more complicated to find if the volume belonged to a template or not and to know which to remove.

Also, it seems the behavior was always to delete all the volumes, the selector just changed at some point and stopped working.